app/scripts/modules/openstack/serverGroup/details/securityGroup/editSecurityGroups.modal.html

```diff
@@ -0,0 +1,35 @@
+<div modal-page>
+  <task-monitor monitor="$ctrl.taskMonitor"></task-monitor>
+  <form role="form" name="form">
+    <modal-close></modal-close>
+    <div class="modal-header">
+      <h3>Edit Security Groups for {{$ctrl.serverGroup.name}}</h3>
+    </div>
+
+    <div class="modal-body container-fluid form-horizontal">
+      <os-app-cache-backed-multi-select-field label="Security Groups"
+                                              cache-key="securityGroups"
+                                              filter="filter"
+                                              required="true"
+                                              model="$ctrl.securityGroups"></os-app-cache-backed-multi-select-field>
+
+      <!--<div class="form-group">-->
+        <!--<div class="col-md-10 col-md-offset-1">-->
+          <!--<ui-select multiple ng-model="$ctrl.command.securityGroups" class="form-control input-sm">-->
+            <!--<ui-select-match>{{$item.name}} ({{$item.id}})</ui-select-match>-->
+            <!--<ui-select-choices repeat="securityGroup as securityGroup in $ctrl.availableSecurityGroups | anyFieldFilter: {name: $select.search, id: $select.search}">-->
+              <!--<span ng-bind-html="securityGroup.name | highlight: $select.search"></span>-->
+              <!--(<span ng-bind-html="securityGroup.id  | highlight: $select.search"></span>)-->
+            <!--</ui-select-choices>-->
+          <!--</ui-select>-->
+        <!--</div>-->
+      <!--</div>-->
+    </div>
+
+    <aws-footer action="$ctrl.submit()"
+                cancel="$ctrl.cancel()"
+                is-valid="$ctrl.isValid()"
+                account="$ctrl.serverGroup.account"
+                verification="$ctrl.state.verification"></aws-footer>
```

Comment:
ok... I'll copy over the required directive.... the other option is to move it to core if there is nothing aws-specific... I'll look into both possibilities.